### ISSUE TITLE
[mediaqueries][mediaqueries-5] Account for CSP restrictions in the scripting feature

### ISSUE THREAD

**Issue by hackademix** [w3c/csswg-drafts#8653]
Regarding the [@media scripting features, value  `none`](https://www.w3.org/TR/mediaqueries-5/#valdef-media-scripting-none), the current specification reads:

> [...]the value of the scripting media feature should be [enabled](https://www.w3.org/TR/mediaqueries-5/#valdef-media-scripting-enabled) or [initial-only](https://www.w3.org/TR/mediaqueries-5/#valdef-media-scripting-initial-only) if scripts originating on the same domain as the document are allowed to run, and [none](https://www.w3.org/TR/mediaqueries-5/#valdef-media-scripting-none) otherwise.

Security and privacy browser extensions such NoScript, uBlock Origin or uMatrix currently use CSP directives (typically `script-src` and `script-src-elem`) to turn off JavaScript on certain sites of users' choice, and use DOM manipulation to display the content of `<noscript>` elements (which are not affected by CSP) and provide users with the authors' intended feedback. Of course the JavaScript-toggled CSS classes approach works out of the box.

It would be great if this new feature took in account CSP disabling script execution on the document, making explicit that the value should be `none` for that case. Otherwise the aforementioned tools would need to scan all the CSS rules applied to the page and manipulate them to force those matching `scripting none`, which would a ugly, inefficient hack.

**Comment by CanadaHonk** [comment]
Personally I like this idea but not sure if it's something that would/should be considered under the spec itself since it's more specifically for the case of external extensions (since no site would need to check if scripting is enabled/disabled if they knew CSP would block it anyway).

**Comment by emilio** [comment]
I think this media feature should match `<noscript>` behavior.

It may be the case that `<noscript>` should do this tho... Right now `<noscript>` uses [this](https://html.spec.whatwg.org/#concept-n-noscript), maybe it should account for `script-src 'none'` or so.

But yeah this discussion maybe should be in https://github.com/whatwg/html/issues/new

**Comment by tabatkins** [comment]
Agreed that the `noscript` element should probably trigger when CSP shuts off scripting for the page (and that's an HTML issue, not a CSS one).

For CSS, the MQ is defined using generic language intentionally - if scripting is disabled, it should match. I do think it's ideal for us to match the `noscript` element, tho, and so UAs should be consistent in that regard.
